tests/Unit/HtmlReplacerPageTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+use function Spatie\Snapshots\assertMatchesTextSnapshot;
+
+it('will not mangle an Empty HTML page', function () {
+    $pageHtml = <<<'HTML'
+<!DOCTYPE html>
+<html lang="en">
+  <head></head>
+  <body></body>
+</html>
+HTML;
+    assertMatchesTextSnapshot(htmlReplacerPngParser($pageHtml));
+});
+
+it('will replace a single Emoji on an page', function () {
+    $pageHtml = <<<'HTML'
+<!DOCTYPE html>
+<html lang="en">
+  <head></head>
+  <body>Hey 🚀</body>
+</html>
+HTML;
+    assertMatchesTextSnapshot(htmlReplacerPngParser($pageHtml));
+});
+
+it('will not replace a single Emoji in the Title', function () {
+    $pageHtml = <<<'HTML'
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta http-equiv="X-UA-Compatible" content="ie=edge">
+        <title>HTML 5🚀 Boilerplate</title>
+        <link rel="stylesheet" href="style.css">
+    </head>
+    <body></body>
+</html>
+HTML;
+    $results = htmlReplacerPngParser($pageHtml);
+    expect($results)->toContain('5🚀')->not()->toContain('&#');
+    assertMatchesTextSnapshot($results);
+});
+
+it('will replace the Emoji on page, but not in head - OLD Charset Method', function () {
+    $pageHtml = <<<'HTML'
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
```

Comment:
This is the "incorrect" method to set charset to UTF8.
Unfortunately it will not be respected by `DOMDocument` and Emoji will become HTML entities.

Reply:
I think this can be fixed before merge - we should just wrap all fragments in our own HTML page that uses the correct charset meta tag. This will ensure they are encoded correctly and then we can capture the fragment from our "template".